portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -0,0 +1,112 @@
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.Comment;
+import com.google.sps.data.CommentList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import java.io.PrintWriter;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Date;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet({"/comment", "/delete-data"}
+)
+public class CommentServlet extends HttpServlet {
+
+    private CommentList comments;
+    private Gson gson = new Gson();
+
+    @Override
+    public void init(){
+        comments = new CommentList();
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Query query = new Query("Comment");//.addSort("timestamp", SortDirection.DESCENDING);
+        PreparedQuery results = datastore.prepare(query);
+        for (Entity entity : results.asIterable()) {
+            Comment comment = new Comment(entity.getProperty("commentName").toString(),
+                entity.getProperty("commentText").toString(),
+                (Date)entity.getProperty("commentDate"));
+            comments.add(comment);
+        }
+    }
+
+    //responds with a comment list from the start of the comments of that page 
+    //to the end of that sublist of comments
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    int numComments = Integer.parseInt(request.getParameter("numComments"));
+    int pageNum = Integer.parseInt(request.getParameter("pageNum"));
+    int start = numComments * (pageNum-1);
+    int end = numComments * pageNum;
+    response.getWriter().print(gson.toJson(comments.getSubList(start,end)));
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    if(request.getServletPath().equalsIgnoreCase("/delete-data")){
+        deleteData();
+    } else {
+        addComment(request, response);
+    }
+  }
```

Comment:
A standard pattern for web pages is POST/REDIRECT/GET (or DELETE/REDIRECT/GET). Read about it more here https://en.wikipedia.org/wiki/Post/Redirect/Get.

So here you'd respond with a redirect back to the same page.

Reply:
Nevermind, HTML5 forms don't support the DELETE verb.